File: src/main/java/com/silverhetch/clotho/connection/TargetImpl.java
package com.silverhetch.clotho.connection;

import javax.net.ssl.SSLSocketFactory;

public class TargetImpl implements Target {
    private final String url;
    private final SSLSocketFactory sslSocketFactory;

    public TargetImpl(String url, SSLSocketFactory sslSocketFactory) {
        this.url = url;
        this.sslSocketFactory = sslSocketFactory;
    }

    public TargetImpl(String url){
        this(url, null);
    }

    @Override
    public String url() {
        return url;
    }

    @Override
    public SSLSocketFactory socketFactory() {
        return sslSocketFactory;
    }
}

File: src/main/java/com/silverhetch/clotho/connection/broadcast/TargetImpl.kt
package com.silverhetch.clotho.connection.broadcast

import java.net.Inet4Address
import java.net.InetAddress
import java.net.NetworkInterface

class TargetImpl(private val networkInterface: NetworkInterface,
                 private val inetAddress: InetAddress) : Target {
    override fun name(): String {
        return networkInterface.displayName
    }

    override fun interfaceInetAddress(): InetAddress {
        val addresses = networkInterface.interfaceAddresses
        addresses.forEach {
            if (it.address is Inet4Address) {
                return it.address
            }
        }
        return InetAddress.getByName("127.0.0.1")
    }

    override fun broadcastAddress(): InetAddress {
        return inetAddress
    }
}